tests/patmat/t12237.check

```diff
@@ -0,0 +1 @@
+21: Pattern Match Exhaustivity: Root, PathAndQuery./(_, _), PathAndQuery.===(_, _), PathAndQuery.:&(_, PathAndQuery.===(_, _)), PathAndQuery.+?(_, PathAndQuery.===(_, _))
```

Comment:
Is this the right expected result?  What does `PathAndQuery./(_, _)` mean? Because the first case is matching values that are `PathAndQuery./`.  Similarly `PathAndQuery.:&(_, PathAndQuery.===(_, _))` and `PathAndQuery.+?(_, PathAndQuery.===(_, _))`.  I don't understand what the check file is asserting.

Reply:
Yes, it's the correct expect file, `PathAndQuery./(_, _)` means that pattern is not exhaustive. The test is to defend against compiler hangs.

Reply:
> `PathAndQuery./(_, _)` means that pattern is not exhaustive.

I see.  (Perhaps reporting `PathAndQuery.:&(_, _)` and `PathAndQuery.+?(_, _)` might be more uniform?)

I guess the Scala 3 checker doesn't try to give good counter-examples like Scala 2, which tries to do something like `Root / "page" / x forSome x != "1"`?

Reply:
We would like to have refined counter-examples as in Scala 2. However, we could not figure out how to make it principled. For example, given the following example:

```Scala
object Even {
  def unapply(x: Int): Option[Int] = ???
}

class Test {
  def foo(x: Option[Option[Int]]): Unit =
    x match {
      case Some(Some(5)) =>
      case Some(Some(Even(10))) =>
      case None =>
    }
}
```

It's unclear how to handle extractors in the warnings. Even if it's possible to be more principled, there's the worry that the message gets messy, thus difficult to read.

Meanwhile, even with refined counter-examples, users still need to check the existing patterns.  Due to the considerations, we only show general counter examples.

Reply:
Yeah, Scala 2's counter-example generation definitely doesn't handle extractors: it's all field based (for case classes) so it has no chance - it needs to be revisited.  But it does a good job for case classes I say.  It seems to be to be a good thing to generalise case class's unapply/extraction with the "product match" extractors, but it seems like discarding that knowledge for counter-example generation is lossy and thus a shame.